public_html/wp-content/mu-plugins/4-helpers-wcpt.php

```diff
@@ -245,16 +245,25 @@ function wcorg_get_wordcamp_duration( WP_Post $wordcamp ) {
  * @param string $name          Optional. The `name` attribute for the `select` element. Defaults to `wordcamp_id`.
  * @param array  $query_options Optional. Extra arguments to pass to `get_posts()`. Defaults to the values in `get_wordcamps()`.
  * @param int    $selected      Optional. The list option to select. Defaults to not selecting any.
+ * @param bool   $cached        Optional. Cache the data. Default false.
  *
  * @return string The HTML for the <select> list.
  */
-function get_wordcamp_dropdown( $name = 'wordcamp_id', $query_options = array(), $selected = 0 ) {
-	$wordcamps = get_wordcamps( $query_options );
+function get_wordcamp_dropdown( $name = 'wordcamp_id', $query_options = array(), $selected = 0, $cached = false ) {
+	$cache_key = 'wc_dropdown_' . md5( serialize( $query_options ) );
+	$cached_data = get_transient( $cache_key );
+
+	if ( $cache_data && $cached ) {
+		$wordcamps = $cached_data;
+	} else {
+		$wordcamps = get_wordcamps( $query_options );
```

Comment:
Do we still risk running out of memory here, because we're initially loading the full results into RAM? It might be safer to only fetch the data we need in the first place.

The `fields` argument of `get_posts()` unfortunately only accepts `ids`, so we'd probably have to use a raw query, but that's fine IMO.

I think that'd also mean we could remove the code below that creates `$wordcamps_cached`.

Reply:
yeah - I did a prototype of calling `get_posts` with just the `ids` parameter, and what happened was less memory but 1900 queries! mainly due to how you access individual fields, and it has to call `get_post` each time - plus the lack of memcache causes high load since the query isn't ever in memory.

if we're not adverse to a custom query, that might be best.

Reply:
oh, I think one thing that was stopping me from a direct query was the `title` field, which if we do a direct query we are unable to apply any title filters that might exist, not sure how important that is?

Reply:
I think skipping the filters would be fine so that we can do a direct query, especially because we're already manually appending the year.